src/libYARP_OS/src/StreamConnectionReader.cpp

```diff
@@ -247,7 +247,9 @@ yarp::os::ConstString StreamConnectionReader::expectText(int terminatingChar)
     }
     yAssert(in!=nullptr);
     bool lsuccess = false;
+    in->setReadTimeout(2.0);
     ConstString result = in->readLine(terminatingChar, &lsuccess);
+    in->setReadTimeout(0.0);
```

Comment:
We should probably get current timeout and reset the previous value

Reply:
Right, I didn't think about that, but now I checked and there is no easy way to get that: no `getReadTimeout()` or `haveReadTimeout()` method in the `InputStream`. Do you think is this the case to add one?